src/copaw/agents/model_factory.py

```
@@ -306,6 +307,86 @@ def _create_formatter_instance(
+            Tuple of (model_instance, formatter_instance)
+        """
+        # Get resolved config for tier
+        resolved = get_model_slot(tier)
+        if resolved is None:
+            # Fall back to active_llm
+            resolved = get_active_llm_config()
+            if resolved is None:
+                raise ValueError(
+                    f"No model configured for tier '{tier}' "
+                    "and no active_llm set",
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)